### PR TITLE
Remove Panel References

### PR DESCRIPTION
**Pull request by WillAyd**
progress towards #25632

### CODE REVIEW

File: pandas/compat/numpy/function.py [hunk 1]
```diff
@@ -288,20 +288,7 @@ def validate_take_with_convert(convert, args, kwargs):
 
 
 def validate_transpose_for_generic(inst, kwargs):
-    try:
```
Comment (WillAyd) [inline pandas/compat/numpy/function.py]: This previously was only modifying the exception message for Panel objects before raising again, so I simplified to just remove the try...except altogether and let it raise as is

File: pandas/core/frame.py [hunk 2]
```diff
@@ -368,9 +368,7 @@ def _constructor(self):
 
     @property
     def _constructor_expanddim(self):
-        # TODO: Raise NotImplementedError or change note in extending.rst
-        from pandas.core.panel import Panel
-        return Panel
+        raise NotImplementedError("Not supported for DataFrames!")
```
Comment (WillAyd) [inline pandas/core/frame.py]: The TODO mentions changing the docs extending.rst but that looks to have already been done

File: pandas/compat/numpy/function.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/compat/numpy/function.py]: If it’s a one liner now, Is this still needed as a separate function?

File: pandas/core/generic.py [hunk 3]
```diff
@@ -4091,8 +4090,7 @@ def sort_values(self, by=None, axis=0, ascending=True, inplace=False,
         0   A    2    0
         1   A    1    1
         """
-        raise NotImplementedError("sort_values has not been implemented "
-                                  "on Panel or Panel4D objects.")
+        raise NotImplementedError("sort_values has not been implemented.")
```
Comment (jbrockmendel) [inline pandas/core/generic.py]: Since this is now never hit, should this be AbstractMethodError?

File: pandas/compat/numpy/function.py (hunk 1 above)
Comment (WillAyd) [reply pandas/compat/numpy/function.py]: Yea could probably just call this directly now; I'll give it a look tonight and repush

File: pandas/core/generic.py (hunk 3 above)
Comment (WillAyd) [reply pandas/core/generic.py]: Yea I think could go for that too

**Comment by jreback** [comment]
an you merge master and some checks are failing

**Comment by jreback** [comment]
thanks @WillAyd love deleting code!

**Comment by jreback** [comment]
https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=11204 

is this because of this PR?

**Comment by WillAyd** [comment]
Yea think so. I'll push a PR soon to correct

**Pull request merged**